daft/ai/transformers/protocols/image_classifier.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, TypedDict
+
+import transformers
+from transformers import pipeline
+from typing_extensions import Unpack
+
+from daft.ai.protocols import ImageClassifier, ImageClassifierDescriptor
+from daft.ai.utils import get_gpu_udf_options, get_torch_device
+
+if TYPE_CHECKING:
+    import numpy as np
+
+    from daft.ai.typing import Label, Options, UDFOptions
+
+
+class TransformersImageClassifierResult(TypedDict):
+    label: str  # labels sorted by likelihood
+    score: float  # probability of each label
+
+
+class TransformersImageClassifierOptions(TypedDict, total=False):
+    batch_size: int | None
```

Comment:
So a user can supply this dictionary and say `batch_size = None`? Or do we want to interpret no specified batch size as meaning None and force a user to give a positive int if they do supply `batch_size`?